### ISSUE TITLE
Acceptance step in ABC algorithm 

### ISSUE THREAD

**Issue by seabbs** [kingaa/pomp#31]
I was reviewing the implementation of [Toni et al's](http://rsif.royalsocietypublishing.org/content/6/31/187) abc-mcmc algorithm and noticed that the acceptance step doesn't appear to include there M5. step. When the priors are uniform I see that alpha=1 and hence the probability of acceptance is 1, does this then mean that the POMP abc function can only be used with uniform priors?  

I may be missing something crucial here or this may be an oversight?

Algorithm outline and code chunks below for information

Thanks for a great package!

![ABC-MCMC algorithm as outlined  in Toni et al, 2009](https://cloud.githubusercontent.com/assets/20317649/21191885/86bc21d2-c21e-11e6-8d92-19937f4dea2e.png)

In the abc function (line 150)
```r

            ## ABC update rule
            distance <- sum(((datval-simval)/scale)^2)
            if( (is.finite(distance)) && (distance<epssq) ){
                theta <- theta.prop
                log.prior <- log.prior.prop
                .accepts <- .accepts+1L
            }
```
and in the pmcmc function 

```r
            ## PMCMC update rule (OK because proposal is symmetric)
            alpha <- exp(pfp.prop@loglik+log.prior.prop-pfp@loglik-log.prior)
            if (runif(1) < alpha) {
                pfp <- pfp.prop
                theta <- theta.prop
                log.prior <- log.prior.prop
                .accepts <- .accepts+1L
}
```

**Comment by kingaa** [comment]
Thanks for the very clear question. The ABC algorithm implemented in `pomp::abc` is the "Likelihood-free MCMC sampler" (Algorithm 3) of [Marin et al. (2012, *Stat. Comp.*)](http://doi.org/10.1007/s11222-011-9288-2) *in the special case that the proposal distribution is symmetric*.  In particular, `pomp::abc` allows the user to supply a proposal distribution (several common choices are also provided), but insists that the distribution be symmetric, i.e., the probability of moving from &theta; to &theta;&ast; must equal that of moving from &theta;&ast; to &theta;.

In the code for `abc`, we have
```
        if (is.finite(log.prior.prop) &&
            runif(1) < exp(log.prior.prop-log.prior)) {
```
which corresponds to line M5 in the Toni et al. exposition, and
```
            distance <- sum(((datval-simval)/scale)^2)
            if( (is.finite(distance)) && (distance<epssq) ){
                theta <- theta.prop
                log.prior <- log.prior.prop
                .accepts <- .accepts+1L
            }
```
which implements the decision in line M4.

To summarize, **the pomp implementation of ABC does not presuppose uniform priors, but it does assume a symmetric proposal**.

Does this help?

**Comment by seabbs** [comment]
Ah yes that clears that up,  thanks for the prompt response and thank you for the paper reference!

**Comment by kingaa** [comment]
P.S. Thanks for the question.  Among other things, you caused me to realize that the Marin paper is missing from the help page.  I've edit the help page to correct this oversight.
